agent/exec/containerd/controller.go

```diff
@@ -195,12 +195,18 @@ func (r *controller) Wait(ctx context.Context) error {
 		return errors.Wrap(err, "inspecting container failed")
 	}
 
+	// TODO(ijc) this shouldn't be needed here, figure out why
+	// .shutdown/.remove are not being called otherwise.
 	shutdownWithExitStatus := func(reason string) error {
 		exitStatus, err := r.adapter.shutdown(ctx)
 		if err != nil {
 			return err
 		}
 		log.G(ctx).Errorf("EXIT STATUS %v", exitStatus)
+		if err := r.adapter.remove(ctx); err != nil {
```

Comment:
Yes, this was still here (nb: it's the same behaviour as current master, since this series also moves the remove from `r.adapter.shutdown` into `r.adapter.remove`) while I figured out the strangeness I mentioned in the PR comment, I worked those out yesterday so I'll clean up those commits and append them here today.